pkg/addonmanager/controllers/agentdeploy/hosted_sync.go

```diff
@@ -93,6 +93,12 @@ func (s *hostedSyncer) sync(ctx context.Context,
 		return addon, nil
 	}
 
+	// waiting for the addon to be deleted when cluster is deleting.
+	// TODO: consider to delete addon in this scenario.
+	if !cluster.DeletionTimestamp.IsZero() {
```

Comment:
so the order of this line of code matters, we must check this after checking addon, right?

A better way is we have a finalize sync separately, which handles those finalize things.